src/integrationTests/options.test.ts

```diff
@@ -209,11 +219,332 @@ describe('useCookie option', () => {
   });
 });
 
+describe('multiple init calls', () => {
+  const identity = makeIdentity();
+  const baseUrl = 'http://test-host';
+  const cookiePath = '/test/';
+  const cookieDomain = 'uidapi.com';
+
+  describe('when nothing has changed', () => {
+    beforeEach(() => {
+      uid2.init({
+        identity: identity,
+        baseUrl: baseUrl,
+        cookiePath: cookiePath,
+      });
+      uid2.init({
+        identity: identity,
+        baseUrl: baseUrl,
+        cookiePath: cookiePath,
+      });
+      uid2.init({
+        identity: identity,
+        baseUrl: baseUrl,
+        cookiePath: cookiePath,
+      });
+    });
+    test('should return next two init calls without changing anything', () => {
+      expect(getUid2LocalStorage().advertising_token).toBe(identity.advertising_token);
+      let storageConfig = getConfigStorage();
+      expect(storageConfig).toBeInstanceOf(Object);
+      expect(storageConfig).toHaveProperty('cookiePath', cookiePath);
+    });
+  });
+
+  describe('new base URL is given', () => {
+    const oldBaseUrl = baseUrl;
+    const newBaseUrl = 'http://example';
+
+    beforeEach(() => {
+      uid2.init({
+        callback: callback,
+        identity: identity,
+        baseUrl: oldBaseUrl,
+        useCookie: true,
+      });
+    });
+    test('should use new base url', () => {
+      uid2.init({
+        baseUrl: newBaseUrl,
+      });
+      const configCookie = getConfigCookie();
+      expect(configCookie).toHaveProperty('baseUrl', newBaseUrl);
+    });
+
+    test('should use old base url', () => {
+      uid2.init({
+        cookiePath: cookiePath,
+      });
+      const configCookie = getConfigCookie();
+      expect(configCookie).toHaveProperty('baseUrl', oldBaseUrl);
+    });
+  });
+
+  describe('new identity provided but expired', () => {
+    const newIdentity = makeIdentity({ refresh_expires: Date.now() - 100000 });
+    const useCookie = true;
+
+    beforeEach(() => {
+      uid2.init({
+        callback: callback,
+        identity: identity,
+        baseUrl: baseUrl,
+        cookiePath: cookiePath,
+        useCookie: useCookie,
+      });
+      uid2.init({
+        identity: newIdentity,
+      });
+    });
+    test('should set value to old identity', () => {
+      expect(getUid2(useCookie).advertising_token).toBe(identity.advertising_token);
+    });
+    test('should set refresh timer once', () => {
+      expect(setTimeout).toHaveBeenCalledTimes(1);
+      expect(clearTimeout).not.toHaveBeenCalled();
+    });
+    test('old ideneity should be in available state', () => {
+      (expect(uid2) as any).toBeInAvailableState(identity.advertising_token);
+    });
+  });
+
+  describe('new identity provided but expires before old identity', () => {
+    const oldIdentity = makeIdentity({ refresh_expires: Date.now() + 5000 });
+    const newIdentity = makeIdentity({ refresh_expires: Date.now() + 100000 });
```

Comment:
Wouldn't the new identity expire after the old identity - so the SDK should use the new identity?

Reply:
The code is checking identity_expres to decide whether to use the new identity or the old one, but you're changing refresh_expires in the test. I think identity_expires is probably the right one to use.

Reply:
Yep - made this fix.  This was part of the tests that seemed to be working because both identities had the same advertising_token